From e7ccdf026961f0a7200ee39b9e426d2ac5ef9f02 Mon Sep 17 00:00:00 2001
From: Anoob Soman <anoob.soman@citrix.com>
Date: Thu, 27 Jul 2017 16:25:54 +0100
Subject: [PATCH] net: Do not scrub ignore_df within the same name space

Commit 239c78db9c ("net: clear local_df when passing skb between namespaces")
cleared local_df/ignore_df for tunnels. This broke, last resort, fragmentation
of GRE tunneled packets.

Clear ignore_df, only when packet is being injected to another namespace.

Signed-off-by: Anoob Soman <anoob.soman@citrix.com>
---
 net/core/skbuff.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/net/core/skbuff.c b/net/core/skbuff.c
index 21a83e26f004..d0c3597f3e6f 100644
--- a/net/core/skbuff.c
+++ b/net/core/skbuff.c
@@ -5857,7 +5857,6 @@ void skb_scrub_packet(struct sk_buff *skb, bool xnet)
 {
 	skb->pkt_type = PACKET_HOST;
 	skb->skb_iif = 0;
-	skb->ignore_df = 0;
 	skb_dst_drop(skb);
 	skb_ext_reset(skb);
 	nf_reset_ct(skb);
@@ -5874,6 +5873,7 @@ void skb_scrub_packet(struct sk_buff *skb, bool xnet)
 
 	skb->mark = 0;
 	skb_clear_tstamp(skb);
+	skb->ignore_df = 0;
 }
 EXPORT_SYMBOL_GPL(skb_scrub_packet);
 
-- 
2.34.1

